mllib/src/main/scala/org/apache/spark/ml/classification/LogisticRegression.scala

```diff
@@ -1447,7 +1447,7 @@ private class LogisticAggregator(
       label: Double): Unit = {
 
     val localFeaturesStd = bcFeaturesStd.value
-    val localCoefficients = bcCoefficients.value
+    val localCoefficients = bcCoefficients.value.toArray
```

Comment:
In the first version of LOR, we have the following code which avoid this issue you pointed out. 

```scala
  private val weightsArray = weights match {
     case dv: DenseVector => dv.values
     case _ =>
       throw new IllegalArgumentException(
         s"weights only supports dense vector but got type ${weights.getClass}.")
   }
```

I think order approach will be more efficient since `toArray` is only called once (you can add the case for sparse), and for sparse initial coefficients, we will not convert from sparse to dense again and again. 

This can be a future work. With L1 applied, the coefficients can be very sparse, so we can compress the coefficients for each iteration, and have specialized implementation for `UpdateInPlace`.